### ISSUE TITLE
Dynamic path with GET request

### ISSUE THREAD

**Issue by jufemaiz** [codegram/hyperclient#84]
Hi all, have a slight conundrum.

Looking at the Hyperclient::Link

https://github.com/codegram/hyperclient/blob/master/lib/hyperclient/link.rb#L84-L91

the _get call on it cannot pass any parameters as params unlike the other options (post, put, and patch)

Where get params are required it cannot be achieved.

**Comment by dblock** [comment]
The link itself needs to be templated, then it can be constructed with parameters. You're not supposed to throw in random parameters into a get link. That said I would be open for allowing code to do that I think. Want to try a PR?

**Comment by jufemaiz** [comment]
@dblock yep, I get that, however there are times when GET parameters are required - even with a RESTful approach. Will give it a crack. Cheers!

**Comment by dblock** [comment]
@bhoggard this is what we discussed for that sort parameter, give it a shot!

**Comment by dblock** [comment]
I took a stab at implementing this and the most efficient solution I found was to patch `URITemplate`. There're a few questions:

1) I don't love it this implementation. We're no longer complying with RFC6570 so monkey-patching is not the way to go. But I don't love rewriting the whole class either.
2) We need to be able to enable/disable this behavior. I wonder whether it needs to be per client, per URI or something else?
3) Everything needs to be documented.
